src/solvers/refinement/string_refinement.cpp

```diff
@@ -911,7 +911,19 @@ static optionalt<exprt> get_array(
 {
   const auto eom = messaget::eom;
   const exprt &size = arr.length();
-  exprt arr_val = simplify_expr(super_get(arr), ns);
+  exprt s = super_get(arr);
+  while(s.id() == ID_if)
+  {
+    if_exprt s_if = to_if_expr(s);
+    simplify(s_if.cond(), ns);
```

Comment:
this copies the whole `s` exprt, would it be better to just copy the condition?

Reply:
and why is this not part of `simplify_expr`?

Reply:
I will change this to `exprt simp_cond = simplify_expr(to_if_expr(s).cond(), ns);` with another `to_if_expr(s).{true,false}_case()` below.